### PR TITLE
Add flag name linter

### PR DESCRIPTION
**Pull request by rafaelrochasilva**
## Implementation Highlights
In order to avoid having flags name with different patterns, a linter was created to help developers use the snake_case format while creating new flags. 

The Rubocop Linter will run looking for IF conditions with Feature Flag names.

### Checklist

Before submitting the PR make sure the following are checked:

- [x] Commit message has a detailed description of what changed and why.
- [x] Tests are added or updated if you fix a bug or add a feature.

### CODE REVIEW

File: lib/rubocop/cop/flag/feature_flag_name.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          (str $_))
+        PATTERN
+
+        RESTRICT_ON_SEND = [:enabled?].freeze
+
+        def on_send(node)
+          feature_flag_name(node) do |name|
+            next if to_snake_case(name) == name
+
+            add_offense(node, message: 'Use snake_case for feature flag names.')
```
Comment (lucasdavila) [inline lib/rubocop/cop/flag/feature_flag_name.rb:28]: can it mention on the error message, that it also need to be lowercase?

Comment (lucasdavila) [reply lib/rubocop/cop/flag/feature_flag_name.rb:28]: Since SCREAMING_SNAKE_CASE can be considered snake case as well 😆

Comment (rafaelrochasilva) [reply lib/rubocop/cop/flag/feature_flag_name.rb:28]: Good catch! ❤️

**Review by lucasdavila — DISMISSED** [review]

**Review by lucasdavila — DISMISSED** [review]

**Review by mateus-lorandi-telus — DISMISSED** [review]

**Review by jose-gomes-junior — CHANGES_REQUESTED** [review]
it is public repo, I would remove mentions to the company, and avoid terms like `respondent`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,15 +15,11 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Setup Ruby
-        uses: actions/setup-ruby@v1
+        uses: ruby/setup-ruby@v1
         with:
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
```
Comment (jose-gomes-junior) [inline .github/workflows/ci.yml:20]:
```suggestion
          bundler-cache: true
```

File: lib/rubocop/cop/flag/feature_flag_name.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Flag
+      # The Feature Flag Name Cop is used to identify flag names that are not
+      # using the sneak case pattern.
```
Comment (jose-gomes-junior) [inline lib/rubocop/cop/flag/feature_flag_name.rb:7]:
```suggestion
      # using the snake case pattern.
```

File: lib/rubocop/cop/flag/feature_flag_name.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            next if to_snake_case(name) == name
+
+            add_offense(node, message: 'Use snake_case with lowercase for feature flag names.')
+          end
+        end
+
+        def to_snake_case(word)
+          word
+            .to_s
+            .gsub(/([A-Z]+)([A-Z][a-z])/, '\1_\2')
```
[the remark this answers is not in the document]
Comment (jose-gomes-junior) [reply lib/rubocop/cop/flag/feature_flag_name.rb:35]: I think that is ok.

**Review by jose-gomes-junior — APPROVED** [review]
one last comment 👍🏼

**Review by jose-gomes-junior — APPROVED** [review]
LGTM

**Pull request merged**